src/control/cmd/daos_admin/handler_test.go

```diff
@@ -348,126 +498,162 @@ func TestDaosAdmin_Handler(t *testing.T) {
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for name, tc := range map[string]struct {
+		req        *pbin.Request
+		bmbc       *bdev.MockBackendConfig
+		expPayload *bdev.FormatResponse
+		expErr     *fault.Fault
+	}{
+		"nil request": {
+			expErr: pbin.PrivilegedHelperRequestFailed("nil request"),
 		},
 		"BdevFormat nil payload": {
 			req: &pbin.Request{
 				Method: "BdevFormat",
 			},
-			expRes: nilPayloadResp,
+			expErr: nilPayloadErr,
 		},
 		"BdevFormat success": {
 			req: &pbin.Request{
 				Method:  "BdevFormat",
 				Payload: bdevFormatReqPayload,
 			},
-			expRes: successResp,
+			expPayload: &bdev.FormatResponse{
+				DeviceResponses: bdev.DeviceFormatResponses{
+					"foo": &bdev.DeviceFormatResponse{
+						Formatted:  true,
+						Controller: &storage.NvmeController{PciAddr: "foo"},
+					},
+				},
+			},
 		},
 		"BdevFormat failure": {
 			req: &pbin.Request{
 				Method:  "BdevFormat",
 				Payload: bdevFormatReqPayload,
 			},
 			bmbc: &bdev.MockBackendConfig{
-				FormatErr: errors.New("format failed"),
+				FormatErr: errors.New("test format failed"),
+			},
+			expPayload: &bdev.FormatResponse{
+				DeviceResponses: bdev.DeviceFormatResponses{
+					"foo": &bdev.DeviceFormatResponse{
+						Error: bdev.FaultFormatError(errors.New("test format failed")),
+					},
+				},
 			},
-			expRes: successResp, // The format error is handled at a higher layer
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
 			log, buf := logging.NewTestLogger(name)
 			defer common.ShowBufferOnFailure(t, buf)
 
-			var destBuf bytes.Buffer
-			sp := scm.NewMockProvider(log, tc.smbc, tc.smsc)
 			bp := bdev.NewMockProvider(log, tc.bmbc)
+			handler := &bdevFormatHandler{bdevHandler: bdevHandler{bdevProvider: bp}}
 
-			gotErr := handleRequest(log, sp, bp, tc.req, &destBuf)
-			common.CmpErr(t, tc.expErr, gotErr)
-
-			if destBuf.Len() > 0 {
-				gotRes := parseResponse(t, &destBuf)
-
-				// We don't need to check the payload details here; we just
-				// want to check that a response came back and that the Error
-				// field was what we expected.
-				wantErr := tc.expRes.Error
-				gotErr := gotRes.Error
-				if gotErr == wantErr {
-					return
-				}
-				if (gotErr == nil || wantErr == nil) ||
-					!strings.Contains(gotErr.Description, wantErr.Description) {
-					t.Fatalf("unexpected error: wanted %s, got %s", wantErr, gotErr)
-				}
-			}
-		})
-	}
-}
+			resp := handler.Handle(log, tc.req)
 
-func TestDaosAdmin_ReadRequest(t *testing.T) {
-	alnum := []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
-	giantPayload := make([]byte, (pbin.MessageBufferSize*5)+1)
-	for i := 0; i < len(giantPayload); i++ {
-		giantPayload[i] = alnum[rand.Intn(len(alnum))]
-	}
-
-	for name, tc := range map[string]struct {
-		req    *pbin.Request
-		expErr error
-	}{
-		"normal payload": {
-			req: &pbin.Request{
-				Method:  "whatever",
-				Payload: []byte(`{"foo":"bar"}`),
-			},
-		},
-		"giant payload": {
```

Comment:
Was this test moved elsewhere? I'd prefer not to drop it entirely, as it validates that we properly handle large payloads.

Reply:
Re-added, and retooled my mocks to play nicely with it. It's in the pbin/app_test.go file now.